SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/OutgoingMessages.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma once
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+enum class EOutgoingMessageType : int32
+{
+	ReserveEntityIdsRequest,
+	CreateEntityRequest,
+	DeleteEntityRequest,
+	ComponentUpdate,
+	CommandRequest,
+	CommandResponse,
+	CommandFailure,
+	LogMessage,
+	ComponentInterest,
+	EntityQueryRequest,
+	Metrics
+};
+
+struct FOutgoingMessage
+{
+	FOutgoingMessage(const EOutgoingMessageType& InType) : Type(InType) {}
+
+	EOutgoingMessageType Type;
+};
+
+struct FReserveEntityIdsRequest : FOutgoingMessage
+{
+	FReserveEntityIdsRequest(const uint32_t& InNumOfEntities)
```

Comment:
Prefer passing primitives by value.
```suggestion
	FReserveEntityIdsRequest(uint32_t InNumOfEntities)
```